src/murmur/Messages.cpp

```diff
@@ -462,8 +464,17 @@ void Server::msgBanList(ServerUser *uSource, MumbleProto::BanList &msg) {
 				b.qdtStart = QDateTime::currentDateTime().toUTC();
 			}
 			b.iDuration = be.duration();
-			if (b.isValid())
+			if (b.isValid()) {
 				qlBans << b;
+				if (! tmpBans.contains(b)) {
+					log(uSource, QString("New or changed ban: %1").arg(b.toString()));
+				}
+			}
+		}
+		foreach(const Ban &tmpBan, tmpBans) {
+			if (! qlBans.contains(tmpBan)) {
+				log(uSource, QString("Removed ban: %1").arg(tmpBan.toString()));
+			}
```

Comment:
I think I'd actually prefer if "tmpBans" was a QSet ("previousBans"?), and once we have qlBans filled out, create a QSet copy if it ("newBans"?).

Then we can simply work out the diff by using the two sets.

QSet<Ban> added = newBans - previousBans;
QSet<Ban> removed = previousBans - newBans;

Then print them. It makes the code clearer, at least to me.

WDYT?